node-notes-cli/app.js

```diff
@@ -0,0 +1,32 @@
+const dataJson = require('./data.json');
+
+const fs = require('fs');
+
+if (process.argv[2] === 'read') {
+  for (const key in dataJson.notes) {
+    console.log(`${key}: ${dataJson.notes[key]}`);
+  }
+}
+
+if (process.argv[2] === 'create') {
+  dataJson.notes[dataJson.nextId] = process.argv[3];
+  dataJson.nextId = parseInt(Object.keys(dataJson.notes)[Object.keys(dataJson.notes).length - 1]) + 1;
+  fs.writeFile('./data.json', JSON.stringify(dataJson, null, 2), err => {
+    if (err) throw err;
+  });
+}
+
+if (process.argv[2] === 'update') {
+  dataJson.notes[parseInt(process.argv[3])] = process.argv[4];
+  fs.writeFile('./data.json', JSON.stringify(dataJson, null, 2), err => {
+    if (err) throw err;
+  });
+}
+
+if (process.argv[2] === 'delete') {
+  delete dataJson.notes[parseInt(process.argv[3])];
+  dataJson.nextId = parseInt(Object.keys(dataJson.notes)[Object.keys(dataJson.notes).length - 1]) + 1;
```

Comment:
The `nextId` should never go backwards. An ID is forever, sort of like a drivers license. If you are sitting at a bar with 3 friends and one of you goes to the restroom, you don't shuffle your ids around until they come back 😄